### PR TITLE
[W5][T08-4]Shanon Seet

### PR DESCRIPTION
**Pull request by shanseet**
Added listall command that shows all private and non-private details of all contacts in the address book.

### CODE REVIEW

**Comment by ssunil3232** [comment]
Clean coding, very good improvisations, overall found it a very good and functioning feature

**Comment by weizhang05** [comment]
Good alternative to the current "list" command. May consider separating it to multiple commits (i.e. 1 for the codes and another for the documentation).

**Review by ssunil3232 — COMMENTED** [review]
A very well constructed and useful command. Clean code.

**Comment by tantantan277** [comment]
Very detailed and thorough commit, kudos for updating every aspects

File: src/seedu/addressbook/commands/ListAllCommand.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public static final String COMMAND_WORD = "listall";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book as a list with index numbers, " +
+            "including private details.\n"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
```
Comment (stephlewyh) [inline src/seedu/addressbook/commands/ListAllCommand.java:21]: Missing header comment. All non-trivial methods should have java doc format header comments.

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (stephlewyh) [inline src/seedu/addressbook/parser/Parser.java:14]: Imported classes should always be listed explicitly. E.g. `java.util.List` and not `java.util.*`, which does not follow NUS Java coding standards.

File: src/seedu/addressbook/ui/TextUi.java [hunk 3]
```diff
@@ -132,6 +132,14 @@ public void showResultToUser(CommandResult result) {
         showToUser(result.feedbackToUser, DIVIDER);
     }
 
+    public void showListAllResultToUser(CommandResult result) {
```
Comment (stephlewyh) [inline src/seedu/addressbook/ui/TextUi.java:135]: Missing header comment. All non-trivial methods should have java doc format header comments.

**Review by stephlewyh — COMMENTED** [review]
Hi Shannon, 

Nice try! You've done the following:
(1) Updated user documentation
(2) Updated tests for new enhancement
(3) Provided demonstrable enhancement

However, you could be mindful with header comments as it was missing from new methods you added. This will reduce code readability. 

You may close this PR now.

**Pull request closed without merging**